genienlp/calibrate.py

```diff
@@ -44,6 +44,58 @@
 logger = logging.getLogger(__name__)
 
 
+def parse_argv(parser):
+    parser.add_argument(
+        '--confidence_path',
+        required=True,
+        type=str,
+        help='The path to the pickle file where the list of ConfidenceFeatures objects is saved',
+    )
+    parser.add_argument(
+        '--eval_metric',
+        type=str,
+        default='aucpr',
+        choices=['aucpr'],
+        help='An xgboost metric. The metric which will be used to select the best model on the validation set.',
+    )
+    parser.add_argument(
+        '--dev_split',
+        type=float,
+        default=0.2,
+        help='The portion of the dataset to use for validation. The rest is used to train.',
+    )
+    parser.add_argument('--seed', type=int, default=123, help='Random seed to use for reproducibility')
+    parser.add_argument(
+        '--save', required=True, type=str, help='The directory to save the calibrator model and plots after training'
+    )
+    parser.add_argument(
+        '--name_prefix', required=True, type=str, help='A string to prepend to files associated with the calibrator.'
+    )
+    parser.add_argument(
+        '--plot', action='store_true', help='If True, will plot metrics and save them. Requires Matplotlib installation.'
+    )
+    parser.add_argument(
+        '--testing',
+        action='store_true',
+        help='If True, will change labels so that not all of them are equal. This is only used for testing purposes.',
+    )
+    parser.add_argument(
+        '--fast',
+        action='store_true',
+        help='If True, will only train calibrators that don\'t use MC dropout. This substantially increases inference speed.',
+    )
+
+    # Options to normalize scores for a given threshold
+    parser.add_argument(
+        '--threshold',
+        type=float,
+        default=None,
+        help='The threshold above (below) which scores are considered to be positive (negative).',
+    )
+    parser.add_argument('--precision', type=float, default=None, help='Set this if scores should be normalize by precision.')
+    parser.add_argument('--recall', type=float, default=None, help='Set this if scores should be normalize by recall.')
```

Comment:
Why did you move this around? Is there any functional change here? It's hard to tell.

Reply:
No functional changes. I just moved the `parse_argv` to the top for predict, server, and calibrate to be consistent with the rest of the code.

Reply:
Next time, it will be helpful to have a dedicated commit for this change (with the magic words "No functional changes") so I won't worry about checking line by line.

Reply:
I had this atomic commit:  "Move parse_argv method to the top consistent with the rest of the code" `10b9d9f0bb0252a8a51ccc026742b354b449785f` 
Good point on using magic words! Will do next time.

Reply:
Uh I swear I read through each commit. My bad, sorry!

Reply:
No worries!